### ISSUE TITLE
Type-checking is not working in some cases

### ISSUE THREAD

**Issue by eladb** [winglang/wing#1069]
### Community Note

> Please vote by adding a 👍 reaction to the issue to help us prioritize.
> If you are interested to work on this issue, please leave a comment.

### I tried this:

```js
bring cloud;

let bucket = new cloud.Bucket();

let b = inflight () => {
  let a = bucket.list();
  let boom = a.lengt92kk39ujdfidijsdkjoi39834;
};
```

### I expected this:

I expected to get a compilation error that tells me that `Array<str>` doesn't have a `lengt92kk39ujdfidijsdkjoi39834` property.

### Instead, this happened:

Compiled successfully

### Component:

Compiler

### Wing Version:

0.4.57

**Comment by eladb** [comment]
@staycoolcall911 this is P1 -- type checking is not working.

**Comment by staycoolcall911** [comment]
This was fixed in #1548
